src/traces/parcoords/defaults.js

```diff
@@ -0,0 +1,101 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Lib = require('../../lib');
+var attributes = require('./attributes');
+var hasColorscale = require('../../components/colorscale/has_colorscale');
+var colorscaleDefaults = require('../../components/colorscale/defaults');
+
+var handleLineDefaults = function lineDefaults(traceIn, traceOut, defaultColor, layout, coerce) {
+
+    coerce('line.color', defaultColor);
+    coerce('line.colorscale');
+
+    if(hasColorscale(traceIn, 'line')) {
+        colorscaleDefaults(traceIn, traceOut, layout, coerce, {prefix: 'line.', cLetter: 'c'});
+    }
+    else {
+        coerce('line.color', defaultColor);
+    }
+};
+
+
+function dimensionsDefaults(traceIn, traceOut) {
+    var dimensionsIn = traceIn.dimensions || [],
+        dimensionsOut = traceOut.dimensions = [];
+
+    var dimensionIn, dimensionOut, i;
+    var commonLength = Infinity;
+
+    dimensionsIn.splice(60); // parcoords supports up to this many dimensions
+
+    function coerce(attr, dflt) {
+        return Lib.coerce(dimensionIn, dimensionOut, attributes.dimensions, attr, dflt);
+    }
+
+    for(i = 0; i < dimensionsIn.length; i++) {
+        dimensionIn = dimensionsIn[i];
+        dimensionOut = {};
+
+        if(!Lib.isPlainObject(dimensionIn) || !Array.isArray(dimensionIn.values)) {
+            continue;
+        }
+
+        coerce('label');
+        coerce('tickvals');
+        coerce('ticktext');
+        coerce('visible');
```

Comment:
I've made the change to the effect that a missing, non-array or empty `dimension.values` leads to `visible === false`. But maybe I don't fully understand this follow-up feedback. My reading is that if `visible` is ultimately false for whatever reason (the user specified `visible: false`, or didn't supply a proper, non-zero length array for `dimension.values`, then we have to `expect(dimensionsOut[i]).toEqual(dimensionsIn[i])` except of course checking the equality of `visible` itself. If this is correct, I need to move `coerce(visible)` to the top so that none of the other coercions kick in at all. 

I'm adding a commit with this interpretation, just wanted to raise the flag that it's a bit speculative on my part. In the current situation, I can safely test if `values.length < 0` because the `dflt` value for `values` is the empty `[]`. But I need to do the more elaborate, pre-coercion test to see if it's missing, or non-array, as even `dimension.values` should be just pass-through if  `dimension.visible === false`.